### DISCUSSION TITLE
external-converters: option is autmatically removed from configuration.yamlevery thime a new device is removed or added to the network

### DISCUSSION THREAD

**Discussion by ok-67** [Koenkk/zigbee2mqtt#19329] [Devices]
My configuration.yaml looks like shown below, but as soon as I add or remove a device from my zigbee network the

external_converters:
  - plugg.js
 
option is removed from the configuration file

Any idea what is wrong???

-----------
homeassistant: true
external_converters:
  - plugg.js
advanced:
  network_key:
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
  pan_id: 9631
  ext_pan_id:
    - x
    - x
    - x
    - x
    - x
    - x
    - x
    - x
  homeassistant_legacy_entity_attributes: false
  legacy_api: false
  legacy_availability_payload: false
  log_syslog:
    app_name: Zigbee2MQTT
    eol: /n
    host: localhost
    localhost: localhost
    path: /dev/log
    pid: process.pid
    port: 514
    protocol: udp4
    type: '5424'
  channel: 25
  log_level: debug
mqtt:
  server: mqtt://core-mosquitto:1883
  user: MQTT-user
  password: *******
serial:
  port: >-
    /dev/serial/by-id/usb-ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_245bc39b5145ed1187c5ca8f0a86e0b4-if00-port0
frontend:
  port: 8099
device_options:
  legacy: false
devices:
  '0x90395efffeff42cb':
    friendly_name: Kjøkken_kaffemaskin_bryter
  '0x001788010caa6a3d':
    friendly_name: Kjøkken_kaffemaskin_plugg
  '0xb4e3f9fffe66a518':
    friendly_name: Kontor_spott_plugg
  '0x0c4314fffea78118':
    friendly_name: Kontor_spott_bryter
  '0x00158d00091e5a3b':
    friendly_name: Stue_måler_temperatur_luftfukt
  '0x3fff00060000006b':
    friendly_name: Namron-ovn
    thermostat_unit: celsius
------------

**Comment by rmazzali** [comment]
the same for me if I add it at the end of the file. If I place it under the ADVANCED section it does not happen BUT the converter is not considered at startup

**Comment by Koenkk** [comment]
You need to stop z2m before adding the ext converter

**Reply by rmazzali** [reply]
do you mean that I need to change the configuration.yaml file only with the service not running?

**Reply by Koenkk** [reply]
yes!

**Reply by rmazzali** [reply]
another point: should the external_converters statement be under the ADVANCED section or not?

**Reply by Koenkk** [reply]
order doesnt matter but it should be outside (no indentation)

**Reply by rmazzali** [reply]
Thank you,
 but if I put it outside when I restart the application it does not start. Log is not generated. Can you have a look to the attached file, please?
[configuration.yaml.txt](https://github.com/Koenkk/zigbee2mqtt/files/13329071/configuration.yaml.txt)

**Reply by Koenkk** [reply]
`external_converters` should be outside `advanced`, like:

```
...
serial:
  port: >-
    /dev/serial/by-id/usb-Silicon_Labs_Sonoff_Zigbee_3.0_USB_Dongle_Plus_0001-if00-port0
external_converters:
    - GWA1502.js
advanced:
  network_key:
    - 84
    - 232
    - 106
    - 43
 ...
```

**Reply by rmazzali** [reply]
ok thank you. How can understand why Zigbee2MQQT does not start if I put it there?
